userspace/base_setup.sh

```diff
@@ -108,6 +107,19 @@ apt-fast install --no-install-recommends -yq \
     wireless-tools \
     zlib1g-dev
 
+# keep LLVM 17 (not 18 as in 24.04, since mesa/rusticl/etc seems to use 17)
+# clang & llvm metapackages install LLVM 18 (which is not installed anymore)
+apt-fast install --no-install-recommends -yq \
+    clang-17 \
+    llvm-17
```

Comment:
I would worry about this later. we really don't want to add hacks now for future concerns. they'll be easy enough to change when we know we need them

Reply:
I don’t like this change, also, but 24.04 mesa depends right now on llvm-17. we can’t remove mesa since we need it for software rendering and also in mainline.

installing llvm 24.04 meta package installs llvm-18.

so, because of this we have both llvm-17 and llvm-18 installed, taking an extra ~150 MB of space.

we can avoid removing and keep both, until mesa updates to llvm-18, and llvm-17 will not get installed anymore (without any changes).

Reply:
@robin-reckmann what’s your take on this? I’m curious about your opinion.